src/lib/Button/Button.tsx

```diff
@@ -0,0 +1,377 @@
+/* eslint-disable import/prefer-default-export */
+import type { ButtonHTMLAttributes } from 'react';
+import React, { forwardRef } from 'react';
+import styled from 'styled-components';
+
+type Fill = 'solid' | 'outline' | 'ghost';
+type Size = 'small' | 'default' | 'large';
+type Variant = 'primary' | 'secondary' | 'affirmative' | 'destructive';
+
+type Props = Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'size'> & {
```

Comment:
@nall-near We're to the point where we should be making some decisions about overall styling/themes for this app -- these were brought in from near-discovery; should they be from a design perspective?